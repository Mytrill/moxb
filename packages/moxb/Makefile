all:
	cd ../.. && $(MAKE)

# this will forward any other make goal to the parent makefile
$(MAKECMDGOALS):
	cd ../.. && $(MAKE) $(MAKECMDGOALS)
