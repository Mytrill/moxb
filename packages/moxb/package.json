{
  "name": "@moxb/moxb",
  "version": "0.1.3",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "build",
  "typings": "build/",
  "files": [
    "build/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "dependencies": {
    "tslib": "^1.9.3"
  },
  "peerDependencies": {
    "mobx": "^4.3.0 || ^5.0.0"
  },
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@types/meteor": "^1.4.16",
    "jest": "^23.3.0",
    "prettier": "^1.13.7",
    "ts-jest": "^23.0.0",
    "typescript": "^2.9.2"
  },
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ]
}
