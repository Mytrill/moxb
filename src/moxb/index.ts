export * from './impl';
export * from './meteor';
export * from './types';
