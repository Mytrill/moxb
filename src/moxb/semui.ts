export * from './action/ActionUi';
export * from './bind/BindUi';
export * from './bool/BoolUi';
export * from './confirm/ConfirmUi';
export * from './many-of/ManyOfUi';
export * from './modal/ModalUi';
export * from './numeric/NumericUi';
export * from './one-of/OneOfUi';
export * from './table/TablePaginationUi';
export * from './table/TableSearchUi';
export * from './table/TableUi';
export * from './text/TextUi';
