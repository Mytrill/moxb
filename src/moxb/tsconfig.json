{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["DOM","ES6","ES2015","ES2017", "DOM.Iterable","ScriptHost"],
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "../build",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false
    },
    "compilerOptions": {
        "baseUrl": "",
        "paths": {
            "*": [
                "*",
                "../../optional-modules/node_modules/*"
            ]
        }
    },
    "exclude": [
        "../node_modules"
    ]
}
