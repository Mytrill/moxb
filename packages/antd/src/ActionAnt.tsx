import { Action } from '@moxb/moxb';
import { Button, Spin } from 'antd';
import { ButtonShape, ButtonSize, ButtonType } from 'antd/lib/button';
import { NativeButtonProps } from 'antd/lib/button/button';
import { observer } from 'mobx-react';
import * as React from 'react';
import { BindAntProps, parseProps } from './BindAnt';
import { BindFormItemAntProps, FormItemAnt, parsePropsForChild } from './FormItemAnt';
import { MouseEvent } from 'react';

export type BindActionAntProps = BindAntProps<Action> & NativeButtonProps;

@observer
export class ActionButtonAnt extends React.Component<BindActionAntProps> {
    protected handleClick(event: MouseEvent<any>) {
        const { operation, stopPropagation } = parseProps(this.props, this.props.operation);

        console.log('event', event);

        if (stopPropagation) {
            event.stopPropagation();
        }

        if (operation.pending) {
            // console.log('Ignoring click on pending operation');
        } else {
            operation.fire();
        }
    }

    constructor(props: BindActionAntProps) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    render() {
        const { operation, invisible, children, label, id, size, shape, htmlType, type, reason, ...props } = parseProps(
            this.props,
            this.props.operation
        );
        if (invisible || operation.invisible) {
            return null;
        }
        return (
            <Button
                id={id}
                onClick={this.handleClick}
                {...props}
                size={size as ButtonSize}
                shape={shape as ButtonShape}
                type={type as ButtonType}
                title={reason}
                htmlType={typeof htmlType === 'undefined' ? 'button' : htmlType}
            >
                {children != null ? children : label}
                {operation.pending && <Spin />}
            </Button>
        );
    }
}

@observer
export class ActionSpanAnt extends React.Component<BindActionAntProps> {
    protected handleClick(event: MouseEvent<any>) {
        const { operation, stopPropagation } = parseProps(this.props, this.props.operation);

        if (stopPropagation) {
            event.stopPropagation();
        }

        operation.fire();
    }

    constructor(props: BindActionAntProps) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    render() {
        const { operation, invisible, children, label, id, reason, stopPropagation, ...props } = parseProps(
            this.props,
            this.props.operation
        );
        if (invisible || operation.invisible) {
            return null;
        }
        return (
            <span id={id} onClick={this.handleClick} title={reason} {...props as any}>
                {children != null ? children : label}
            </span>
        );
    }
}

@observer
export class ActionFormButtonAnt extends React.Component<BindActionAntProps & BindFormItemAntProps> {
    render() {
        const { operation, invisible, ...props } = parsePropsForChild(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <FormItemAnt operation={operation} label={null} {...this.props as any}>
                <ActionButtonAnt htmlType="submit" operation={operation} {...props} />
            </FormItemAnt>
        );
    }
}
