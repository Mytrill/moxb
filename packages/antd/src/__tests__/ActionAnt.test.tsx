import { ActionImpl } from '@moxb/moxb';
import * as React from 'react';
import { ActionButtonAnt } from '../ActionAnt';
const ReactTestRenderer = require('react-test-renderer');

describe('ActionFormButtonAnt', function() {
    it('should render', function() {
        const operation = new ActionImpl({
            id: 'TheId',
            fire: jest.fn(),
            label: 'aha',
            disabled: () => true,
        });

        expect(ReactTestRenderer.create(<ActionButtonAnt operation={operation} />)).toMatchSnapshot();
    });
});
