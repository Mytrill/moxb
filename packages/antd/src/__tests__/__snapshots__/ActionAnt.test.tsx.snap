// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButtonAnt should render a button by default 1`] = `
<Button
  block={false}
  disabled={true}
  ghost={false}
  htmlType="button"
  id="the_id"
  loading={false}
  onClick={[Function]}
  prefixCls="ant-btn"
  readOnly={false}
>
  aha
</Button>
`;

exports[`ActionButtonAnt should render a button with children instead of a label 1`] = `
<Button
  block={false}
  disabled={false}
  ghost={false}
  htmlType="button"
  id="the_id"
  loading={false}
  onClick={[Function]}
  prefixCls="ant-btn"
  readOnly={false}
>
  FooButton
</Button>
`;

exports[`ActionFormButtonAnt should render a button by default 1`] = `
<FormItemAnt
  label={null}
  operation={
    bound ActionImpl {
      "_errors": Array [],
      "constructor": [Function],
      "doClearErrors": [Function],
      "doOnValidate": [Function],
      "domId": "the_id",
      "getDisabled": [Function],
      "getErrors": [Function],
      "getHelp": [Function],
      "getInvisible": [Function],
      "getLabel": [Function],
      "getPending": [Function],
      "getReadonly": [Function],
      "getReason": [Function],
      "id": "TheId",
      "impl": Object {
        "disabled": [Function],
        "fire": [Function],
        "id": "TheId",
        "label": "aha",
      },
      "setError": [Function],
      Symbol(mobx did run lazy initializers): true,
      Symbol(mobx administration): ObservableObjectAdministration$$1 {
        "defaultEnhancer": [Function],
        "keysAtom": Atom$$1 {
          "diffValue": 0,
          "isBeingObserved": false,
          "isPendingUnobservation": false,
          "lastAccessedBy": 0,
          "lowestObserverState": 2,
          "name": "ActionImpl@61.keys",
          "observers": Set {},
        },
        "name": "ActionImpl@61",
        "target": [Circular],
        "values": Map {
          "_errors" => Array [],
          "label" => "aha",
          "disabled" => true,
          "enabled" => false,
          "reason" => undefined,
          "invisible" => false,
          "readOnly" => false,
          "help" => undefined,
          "errors" => Array [],
          "error" => undefined,
          "hasErrors" => false,
        },
      },
    }
  }
>
  <ActionButtonAnt
    disabled={true}
    htmlType="submit"
    id="the_id"
    operation={
      bound ActionImpl {
        "_errors": Array [],
        "constructor": [Function],
        "doClearErrors": [Function],
        "doOnValidate": [Function],
        "domId": "the_id",
        "getDisabled": [Function],
        "getErrors": [Function],
        "getHelp": [Function],
        "getInvisible": [Function],
        "getLabel": [Function],
        "getPending": [Function],
        "getReadonly": [Function],
        "getReason": [Function],
        "id": "TheId",
        "impl": Object {
          "disabled": [Function],
          "fire": [Function],
          "id": "TheId",
          "label": "aha",
        },
        "setError": [Function],
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "ActionImpl@61.keys",
            "observers": Set {},
          },
          "name": "ActionImpl@61",
          "target": [Circular],
          "values": Map {
            "_errors" => Array [],
            "label" => "aha",
            "disabled" => true,
            "enabled" => false,
            "reason" => undefined,
            "invisible" => false,
            "readOnly" => false,
            "help" => undefined,
            "errors" => Array [],
            "error" => undefined,
            "hasErrors" => false,
          },
        },
      }
    }
    readOnly={false}
  />
</FormItemAnt>
`;
